src/Adapter/Security/Admin.php

```diff
@@ -39,17 +39,26 @@
  */
 class Admin
 {
+    /**
+     * @var LegacyContext
+     */
     private $context;
+
+    /**
+     * @var \Context
+     */
     private $legacyContext;
+
+    /**
+     * @var TokenStorage
+     */
     private $securityTokenStorage;
-    private $userProvider;
 
     /**
-     * Constructor.
-     *
-     * @param LegacyContext $context
-     * @param TokenStorage $securityTokenStorage
+     * @var UserProviderInterface
      */
+    private $userProvider;
+
```

Comment:
Why removing Constructor doc block?

Reply:
Because it adds no value. I've tried with my IDE we have exactly the same information with or without it (try with a DECENT IDE).

More, I've found a lot - like A LOT - of constructor PHPDoc are outdated with their current definition: in this case, PHPStorm gives the information that comes from PHPDocs => not cool.

So, my point is, if we have only arguments that we can type hint, we should avoid using PHPDocs for the constructor function.

Reply:
Hahaha, i've the best IDE ever mate :P If you try you will love it!